storage/storage.c

```diff
@@ -579,6 +582,57 @@ static void derive_kek_v4(const uint8_t *pin, size_t pin_len,
   memzero(&salt, sizeof(salt));
 }
 #endif
+#endif
+
```

Comment:
Below we are duplicating two functions consisting of some 70 lines of code just because of a difference in the usage of one single constant `SECURE_AES_KEY_XORK_SP` vs. `SECURE_AES_KEY_XORK_SN`. AFAICS there is no difference in the procedures themselves. I think that much code duplication is unjustifiable in this case. This can be cleanly solved, for example, by adding a `bool privileged_bhk` parameter or `secure_aes_keysel_t saes_key` parameter to `stretch_pin()` and to `derive_kek_unlock()`.

Reply:
fixed https://github.com/trezor/trezor-firmware/commit/b30fd7697549daa3ce7610007fd0ac3f8923035d

disadvantage is that we have the new parameter on newer (V6 only) models too and its non-functional, as there is no way to use the unpriv key.